### PR TITLE
POST version of GetValidators and GetValidatorBalances

### LINKED ISSUE #13187: BEACON API: add POST validators endpoints for validators and validator_balances endpoints
# 💎 Issue

### Description

Add a `POST` variant for the following endpoints. read more reasoning from  https://github.com/ethereum/beacon-APIs/pull/367

- [ ] `/eth/v1/beacon/states/{state_id}/validators`
- [ ] `/eth/v1/beacon/states/{state_id}/validator_balances`

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

Implements https://github.com/ethereum/beacon-APIs/pull/367. I also simplified testing to use less validators.

**Which issues(s) does this PR fix?**

Fixes #13187

### CODE REVIEW

File: beacon-chain/rpc/eth/beacon/handlers_validator.go [hunk 1]
```diff
@@ -49,7 +51,21 @@ func (s *Server) GetValidators(w http.ResponseWriter, r *http.Request) {
 	}
 	isFinalized := s.FinalizationFetcher.IsFinalized(ctx, blockRoot)
 
-	rawIds := r.URL.Query()["id"]
+	var rawIds []string
+	if r.Method == http.MethodGet {
+		rawIds = r.URL.Query()["id"]
+	} else {
+		err = json.NewDecoder(r.Body).Decode(&rawIds)
```
Comment (james-prysm) [inline beacon-chain/rpc/eth/beacon/handlers_validator.go:58]: I saw the spec and this list should be a unique list of IDs, i don't think decodeIds is checking for uniqueness.

Comment (rkapka) [reply beacon-chain/rpc/eth/beacon/handlers_validator.go:58]: I don't see it anywhere in the spec. It is a useful thing to add, but I would prefer to have it in another PR.

File: beacon-chain/rpc/eth/beacon/handlers_validator.go [hunk 2]
```diff
@@ -72,7 +93,12 @@ func (s *Server) GetValidators(w http.ResponseWriter, r *http.Request) {
 	epoch := slots.ToEpoch(st.Slot())
 	allBalances := st.Balances()
 
-	statuses := r.URL.Query()["status"]
+	var statuses []string
+	if r.Method == http.MethodGet {
```
Comment (nalepae) [inline beacon-chain/rpc/eth/beacon/handlers_validator.go:97]: Maybe grouping `rawIds` and `statuses` to reduce the number of `r.Method == http.MethodGet`?
(Not sure what is the most idiomatic in that case.)

Like:
```go
	var rawIds []string
	var statuses []string

	if r.Method == http.MethodGet {
		rawIds = r.URL.Query()["id"]
		statuses = r.URL.Query()["status"]
	} else {
		rawIds = req.Ids
		statuses = req.Statuses
	}
```

File: beacon-chain/rpc/eth/beacon/handlers_validator.go [hunk 3]
```diff
@@ -49,7 +51,26 @@ func (s *Server) GetValidators(w http.ResponseWriter, r *http.Request) {
 	}
 	isFinalized := s.FinalizationFetcher.IsFinalized(ctx, blockRoot)
 
-	rawIds := r.URL.Query()["id"]
+	var req GetValidatorsRequest
+	if r.Method == http.MethodPost {
+		err = json.NewDecoder(r.Body).Decode(&req)
+		switch {
+		case err == io.EOF:
```
Comment (nalepae) [inline beacon-chain/rpc/eth/beacon/handlers_validator.go:58]: Those errors case seem not to be tested.
(Not sure about the testing policy in that case.)

File: beacon-chain/rpc/eth/beacon/handlers_validator.go [hunk 4]
```diff
@@ -234,7 +260,21 @@ func (bs *Server) GetValidatorBalances(w http.ResponseWriter, r *http.Request) {
...
 	isFinalized := bs.FinalizationFetcher.IsFinalized(ctx, blockRoot)
 
-	rawIds := r.URL.Query()["id"]
+	var rawIds []string
+	if r.Method == http.MethodGet {
+		rawIds = r.URL.Query()["id"]
+	} else {
+		err = json.NewDecoder(r.Body).Decode(&rawIds)
+		switch {
+		case err == io.EOF:
```
Comment (nalepae) [inline beacon-chain/rpc/eth/beacon/handlers_validator.go:269]: Those errors case seem not to be tested.
(Not sure about the testing policy in that case.)

**Review by nalepae — APPROVED** [review]
3 minors comments

File: beacon-chain/rpc/eth/beacon/handlers_validator.go (hunk 2 above)
Comment (rkapka) [reply beacon-chain/rpc/eth/beacon/handlers_validator.go:97]: done

File: beacon-chain/rpc/eth/beacon/handlers_validator.go (hunk 3 above)
Comment (rkapka) [reply beacon-chain/rpc/eth/beacon/handlers_validator.go:58]: done

File: beacon-chain/rpc/eth/beacon/handlers_validator.go (hunk 4 above)
Comment (rkapka) [reply beacon-chain/rpc/eth/beacon/handlers_validator.go:265]: done

**Review by nalepae — APPROVED** [review]

**Pull request merged**